examples/inverse_kinematics/README.md

```diff
@@ -0,0 +1,202 @@
+
+The given Python script will:
+- undock Spot;
+- move it forward approximatively 1m;
+- rotate it 90 degrees counter-clockwise;
+- search and test IK solutions for random tool poses;
+- dock Spot back to the given docking station.
+
+## Running the Example
+
+1. Make sure the robot is docked, and there is approximately 2 meters free in front of the docking station.
+
+2. Make sure you've built and sourced your workspace:
+   ```bash
+   cd <ros2 workspace>
+   colcon build --symlink-install
+   source /opt/ros/humble/setup.bash
+   source ./install/local_setup.bash
+   ```
+3.  Start the driver:
+    ```bash
+    bdai spot driver spot_name --launch-rviz
+    ```
+4.  Run the script:
+    ```bash
+    ros2 run inverse_kinematics send_inverse_kinematics_requests --robot spot_name --dock docking_station_id --poses n
+    ```
+    For example:
+    ```bash
+    ros2 run inverse_kinematics send_inverse_kinematics_requests --robot Opal --dock 527 --poses 10
+    ```
+
+    The following parameters are mandatory:
+
+    ```
+    --robot robot_name
+      The name of the Spot robot to use, e.g. "Opal".
+
+    --dock docking_station_id
+      The docking station where spot is docked, e.g. 527.
+    ```
+    The following parameter is optional:
+
+    ```
+    --poses n
+      The number of tool poses to check (default 1).
+    ```
+
+    The same command can be run directly in VSCode by adding the following to your launch configuration in `launch.json`.
+
+    ```json
+    {
+        "name": "send_inverse_kinematics_requests",
+        "type": "python",
+        "request": "launch",
+        "program": "${workspaceFolder}/ws/src/external/spot_ros2/examples/inverse_kinematics/inverse_kinematics/send_inverse_kinematics_requests.py",
+        "args": [
+            "--robot",
+            "Opal",
+            "--dock",
+            "527",
+            "--poses",
+            "10"
+        ],
+        "console": "integratedTerminal",
+        "justMyCode": true
+    }
+    ```
+
+## Converting Direct API Calls to ROS2
+
+While the driver provides plenty of helper services and topics, direct Spot API calls can also usually be replaced with ROS2 calls simply by converting the protobuf into a ROS message and using the `robot_command` action. This example shows how to update the [Inverse Kinematics Service](https://dev.bostondynamics.com/python/examples/inverse_kinematics/readme) to use ROS2 instead of direct API calls.  The same concepts can be applied to any code that uses the `RobotCommand` protobuf.
+
+We'll go through the changes the [ROS2 code](inverse_kinematics/send_inverse_kinematics_requests.py) makes to the [original example provided by Boston Dynamics](https://github.com/boston-dynamics/spot-sdk/blob/master/python/examples/inverse_kinematics/reachability.py).
+
+The main difference in this example is that it publishes TFs to RViz, instead of generating a plot, to show the expected tool positions respective to the tool frame.
+
+
+TODO
```

Comment:
^

Reply:
Yes, the example document has not been completed yet. Working on it now.

Reply:
Completed.